libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityChecker.cs

```diff
@@ -0,0 +1,372 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AdaptiveCards;
+using Microsoft.Bot.Builder.LanguageGeneration;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Generators
+{
+    public static class ActivityChecker
+    {
+        /// <summary>
+        /// check the lg string result before generate an Activity.
+        /// </summary>
+        /// <param name="lgStringResult">string result from languageGenerator.</param>
+        /// <returns>Diagnostic list.</returns>
+        public static List<Diagnostic> Check(string lgStringResult)
+        {
+            JObject lgStructuredResult;
+            try
+            {
+                lgStructuredResult = JObject.Parse(lgStringResult);
+            }
+            catch
```

Comment:
Is there a better option that using Exceptions for flow control?

Reply:
Currently there is no better way to fix this issue. The output of ILanguageGenerator is string, but we wang to encode some json object into it. So serialization/deserialization is the only way

Reply:
What is good is to add a check if (lgStringResult..Trim()StartsWith("{") && lgStringResult.Trim().EndsWith("}")) then you can avoid the throw in 99% of the time

---
In reply to: [343443844](https://github.com/microsoft/botbuilder-dotnet/pull/2914#discussion_r343443844) [](ancestors = 343443844)